compiler/x/codegen/ControlFlowEvaluator.cpp

```diff
@@ -524,103 +524,98 @@ TR::Register *OMR::X86::TreeEvaluator::tableEvaluator(TR::Node *node, TR::CodeGe
-      generateRegRegInstruction(move_op, node, trgReg->getHighOrder(), reg->getHighOrder(), cg);
-      generateRegRegInstruction(move_op, node, trgReg->getLowOrder(), reg->getLowOrder(), cg);
-
-      dep = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, 4, cg->trMemory());
-      dep->addPostCondition(trgReg->getLowOrder(), TR::RealRegister::NoReg, cg);
-      dep->addPostCondition(trgReg->getHighOrder(), TR::RealRegister::NoReg, cg);
-      dep->addPostCondition(reg->getLowOrder(), TR::RealRegister::NoReg, cg);
-      dep->addPostCondition(reg->getHighOrder(), TR::RealRegister::NoReg, cg);
-      }
-   else
+static TR::Register *minmaxEvaluator64BitPrimitiveOnIA32(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR_ASSERT(TR::Compiler->target.is32Bit(), "ONLY 32-BIT CODE SHOULD REACH HERE.");
+   TR_X86OpCodes SETcc = BADIA32Op;
+   switch (node->getOpCodeValue())
+      {
+      case TR::lmin:
+         SETcc = SETL1Reg;
+         break;
+      case TR::lmax:
+         SETcc = SETG1Reg;
+         break;
+      default:
+         TR_ASSERT(false, "INCORRECT IL OPCODE.");
+         break;
+      }
+   auto oprand0 = cg->evaluate(node->getChild(0));
+   auto oprand1 = cg->evaluate(node->getChild(1));
+   auto result = cg->allocateRegisterPair(cg->allocateRegister(), cg->allocateRegister());
+
+   generateRegRegInstruction(CMP4RegReg, node, oprand0->getLowOrder(), oprand1->getLowOrder(), cg);
+   generateRegInstruction(SETcc, node, result->getLowOrder(), cg); // t1 = (low0 < low1)
+   generateRegRegInstruction(CMP4RegReg, node, oprand0->getHighOrder(), oprand1->getHighOrder(), cg);
+   generateRegInstruction(SETcc, node, result->getHighOrder(), cg); // t2 = (high0 < high1)
+   generateRegRegInstruction(CMOVE4RegReg, node, result->getHighOrder(), result->getLowOrder(), cg); // if (high0 == high1) then t2 = t1 = (low0 < low1)
+
+   generateRegRegInstruction(TEST1RegReg,  node, result->getHighOrder(), result->getHighOrder(),  cg);
+   generateRegRegInstruction(MOV4RegReg,   node, result->getLowOrder(),  oprand0->getLowOrder(),  cg);
+   generateRegRegInstruction(MOV4RegReg,   node, result->getHighOrder(), oprand0->getHighOrder(), cg);
+   generateRegRegInstruction(CMOVE4RegReg, node, result->getLowOrder(),  oprand1->getLowOrder(),  cg);
+   generateRegRegInstruction(CMOVE4RegReg, node, result->getHighOrder(), oprand1->getHighOrder(), cg);
+   node->setRegister(result);
+   for (int i = 0; i < node->getNumChildren(); i++)
       {
-      generateRegRegInstruction(cmp_op, node, reg, trgReg, cg);
-      generateLabelInstruction(isMin ? JG4 : JL4, node, doneLabel, false, cg);
-      generateRegRegInstruction(move_op, node, trgReg, reg, cg);
-
-      dep = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, 2, cg->trMemory());
-      dep->addPostCondition(trgReg, TR::RealRegister::NoReg, cg);
-      dep->addPostCondition(reg, TR::RealRegister::NoReg, cg);
+      cg->decReferenceCount(node->getChild(i));
```

Comment:
Is a loop necessary?  Can `min` or `max` ever have anything other than 2 children?  I'd say just do two `decReferenceCount`s to make the code easier to read.

Reply:
Sure, I've updated this part to two `decReferenceCount`